multimodal/src/autogluon/multimodal/data/infer_types.py

```diff
@@ -690,61 +727,35 @@ def infer_problem_type_output_shape(
-            )
-            if provided_problem_type == BINARY and class_num != 2:
+        return None
+
+    if problem_type in [BINARY, MULTICLASS, REGRESSION]:
+        class_num = len(data[label_column].unique())
+        err_msg = (
+            f"Problem type is '{problem_type}' while the number of "
+            f"unique values in the label column is {class_num}."
+        )
+        if problem_type == BINARY:
+            if class_num != 2:
                 raise AssertionError(err_msg)
-            elif provided_problem_type == MULTICLASS and class_num <= 2:
+            return 2
+        elif problem_type == MULTICLASS:
+            if class_num <= 2:
                 raise AssertionError(err_msg)
-            return provided_problem_type, class_num
-        if provided_problem_type == BINARY:
-            return BINARY, 2
-        elif provided_problem_type == MULTICLASS:
-            class_num = len(data[label_column].value_counts())
-            return MULTICLASS, class_num
-        elif provided_problem_type == CLASSIFICATION:
-            class_num = len(data[label_column].value_counts())
-            if class_num == 2:
-                return BINARY, 2
-            else:
-                return MULTICLASS, class_num
-        elif provided_problem_type == REGRESSION:
-            return provided_problem_type, 1
-        elif provided_problem_type == NER:
-            return provided_problem_type, None
-        elif provided_problem_type == OBJECT_DETECTION:
-            return provided_problem_type, None
-        elif provided_problem_type == OPEN_VOCABULARY_OBJECT_DETECTION:
-            return provided_problem_type, None
-        else:
-            raise ValueError(
-                f"Problem type '{provided_problem_type}' doesn't have a valid output shape "
-                f"for training. The supported problem types are"
-                f" '{BINARY}', '{MULTICLASS}', '{REGRESSION}',"
-                f" '{CLASSIFICATION}', '{NER}',"
-                f" '{OBJECT_DETECTION}', '{OPEN_VOCABULARY_OBJECT_DETECTION}'"
-            )
+            return class_num
+        elif problem_type == REGRESSION:
+            if class_num <= 1:  # in case users want to try toy datasets
+                raise AssertionError(err_msg)
+            return 1
+    elif problem_type in [NER, OBJECT_DETECTION, OPEN_VOCABULARY_OBJECT_DETECTION]:
+        return None
     else:
-        if column_types[label_column] == CATEGORICAL:
-            class_num = len(data[label_column].unique())
-            if class_num == 2:
-                return BINARY, 2
-            else:
-                return MULTICLASS, class_num
-        elif column_types[label_column] == NUMERICAL:
-            return REGRESSION, 1
-        else:
-            raise ValueError(
-                f"The label column '{label_column}' has type"
-                f" '{column_types[label_column]}', which is not supported yet."
-            )
+        raise ValueError(
```

Comment:
It might be confusing if we support `CLASSIFICATION` as input problem type but not include it in the supported problem types here.

Reply:
Added `CLASSIFICATION ` back.